File: webapps/js/ajax/Effects.js
// Provide a default path to dwr.engine
if (dwr == null) var dwr = {};
if (dwr.engine == null) dwr.engine = {};
if (DWREngine == null) var DWREngine = dwr.engine;

dwr.engine._defaultPath = '/dwr';

if (Effects == null) var Effects = {};
Effects._path = '/dwr';
Effects.initSession = function(callback) {
  dwr.engine._execute(Effects._path, 'Effects', 'initSession', callback);
}
Effects.fadeEffect = function(p0, callback) {
  dwr.engine._execute(Effects._path, 'Effects', 'fadeEffect', p0, callback);
}
Effects.fadeEffect = function(p0, p1, callback) {
  dwr.engine._execute(Effects._path, 'Effects', 'fadeEffect', p0, p1, callback);
}
Effects.blindUpEffect = function(p0, callback) {
  dwr.engine._execute(Effects._path, 'Effects', 'blindUpEffect', p0, callback);
}
Effects.blindUpEffect = function(p0, p1, callback) {
  dwr.engine._execute(Effects._path, 'Effects', 'blindUpEffect', p0, p1, callback);
}
Effects.blindDownEffect = function(p0, callback) {
  dwr.engine._execute(Effects._path, 'Effects', 'blindDownEffect', p0, callback);
}
Effects.blindDownEffect = function(p0, p1, callback) {
  dwr.engine._execute(Effects._path, 'Effects', 'blindDownEffect', p0, p1, callback);
}
Effects.appearEffect = function(p0, callback) {
  dwr.engine._execute(Effects._path, 'Effects', 'appearEffect', p0, callback);
}
Effects.appearDownEffect = function(p0, p1, callback) {
  dwr.engine._execute(Effects._path, 'Effects', 'appearDownEffect', p0, p1, callback);
}
Effects.dropOutEffect = function(p0, callback) {
  dwr.engine._execute(Effects._path, 'Effects', 'dropOutEffect', p0, callback);
}
Effects.dropOutEffect = function(p0, p1, callback) {
  dwr.engine._execute(Effects._path, 'Effects', 'dropOutEffect', p0, p1, callback);
}
Effects.foldEffect = function(p0, callback) {
  dwr.engine._execute(Effects._path, 'Effects', 'foldEffect', p0, callback);
}
Effects.foldDownEffect = function(p0, p1, callback) {
  dwr.engine._execute(Effects._path, 'Effects', 'foldDownEffect', p0, p1, callback);
}
Effects.growEffect = function(p0, callback) {
  dwr.engine._execute(Effects._path, 'Effects', 'growEffect', p0, callback);
}
Effects.growEffect = function(p0, p1, callback) {
  dwr.engine._execute(Effects._path, 'Effects', 'growEffect', p0, p1, callback);
}
Effects.highlightEffect = function(p0, callback) {
  dwr.engine._execute(Effects._path, 'Effects', 'highlightEffect', p0, callback);
}
Effects.hightlightEffect = function(p0, p1, callback) {
  dwr.engine._execute(Effects._path, 'Effects', 'hightlightEffect', p0, p1, callback);
}
Effects.puffEffect = function(p0, callback) {
  dwr.engine._execute(Effects._path, 'Effects', 'puffEffect', p0, callback);
}
Effects.puffEffect = function(p0, p1, callback) {
  dwr.engine._execute(Effects._path, 'Effects', 'puffEffect', p0, p1, callback);
}
Effects.pulsateEffect = function(p0, callback) {
  dwr.engine._execute(Effects._path, 'Effects', 'pulsateEffect', p0, callback);
}
Effects.pulsateEffect = function(p0, p1, callback) {
  dwr.engine._execute(Effects._path, 'Effects', 'pulsateEffect', p0, p1, callback);
}

File: Java/com/verilion/object/jsremoting/Effects.java
//------------------------------------------------------------------------------
//Copyright (c) 2003-2005 Verilion Inc.
//------------------------------------------------------------------------------
//Created on 2005-05-28
//Revisions
//------------------------------------------------------------------------------
//$Log: Effects.java,v $
//Revision 1.1.2.1  2007/02/04 00:58:48  tcs
//Initial entry
//
//------------------------------------------------------------------------------
package com.verilion.object.jsremoting;

import org.directwebremoting.WebContext;
import org.directwebremoting.WebContextFactory;
import org.directwebremoting.proxy.scriptaculous.Effect;

/**
 * Use DWR to invoke scriptaculous effects on a single browser's web page
 * 
 * @author Trevor
 * 
 */
public class Effects {

	public Effects() {

	}

	public void initSession() {
		WebContext wctx = WebContextFactory.get();
		wctx.getScriptSession();
	}

	public void fadeEffect(String id) {
		WebContext wctx = WebContextFactory.get();
		Effect effect = new Effect(wctx.getScriptSession());
		effect.fade(id);
	}

	public void fadeEffect(String id, String options) {
		WebContext wctx = WebContextFactory.get();
		Effect effect = new Effect(wctx.getScriptSession());
		effect.fade(id, "{" + options + "}");
	}

	public void blindUpEffect(String id) {
		WebContext wctx = WebContextFactory.get();
		Effect effect = new Effect(wctx.getScriptSession());
		effect.blindUp(id);
	}

	public void blindUpEffect(String id, String options) {
		WebContext wctx = WebContextFactory.get();
		Effect effect = new Effect(wctx.getScriptSession());
		effect.blindUp(id, "{" + options + "}");
	}

	public void blindDownEffect(String id) {
		WebContext wctx = WebContextFactory.get();
		Effect effect = new Effect(wctx.getScriptSession());
		effect.blindDown(id);
	}

	public void blindDownEffect(String id, String options) {
		WebContext wctx = WebContextFactory.get();
		Effect effect = new Effect(wctx.getScriptSession());
		effect.blindDown(id, "{" + options + "}");
	}

	public void appearEffect(String id) {
		WebContext wctx = WebContextFactory.get();
		Effect effect = new Effect(wctx.getScriptSession());
		effect.appear(id);
	}

	public void appearDownEffect(String id, String options) {
		WebContext wctx = WebContextFactory.get();
		Effect effect = new Effect(wctx.getScriptSession());
		effect.appear(id, "{" + options + "}");
	}

	public void dropOutEffect(String id) {
		WebContext wctx = WebContextFactory.get();
		Effect effect = new Effect(wctx.getScriptSession());
		effect.dropOut(id);
	}

	public void dropOutEffect(String id, String options) {
		WebContext wctx = WebContextFactory.get();
		Effect effect = new Effect(wctx.getScriptSession());
		effect.dropOut(id, "{" + options + "}");
	}

	public void foldEffect(String id) {
		WebContext wctx = WebContextFactory.get();
		Effect effect = new Effect(wctx.getScriptSession());
		effect.fold(id);
	}

	public void foldDownEffect(String id, String options) {
		WebContext wctx = WebContextFactory.get();
		Effect effect = new Effect(wctx.getScriptSession());
		effect.fold(id, "{" + options + "}");
	}

	public void growEffect(String id) {
		WebContext wctx = WebContextFactory.get();
		Effect effect = new Effect(wctx.getScriptSession());
		effect.grow(id);
	}

	public void growEffect(String id, String options) {
		WebContext wctx = WebContextFactory.get();
		Effect effect = new Effect(wctx.getScriptSession());
		effect.grow(id, "{" + options + "}");
	}

	public void highlightEffect(String id) {
		WebContext wctx = WebContextFactory.get();
		Effect effect = new Effect(wctx.getScriptSession());
		effect.highlight(id);
	}

	public void hightlightEffect(String id, String options) {
		WebContext wctx = WebContextFactory.get();
		Effect effect = new Effect(wctx.getScriptSession());
		effect.highlight(id, "{" + options + "}");
	}

	public void puffEffect(String id) {
		WebContext wctx = WebContextFactory.get();
		Effect effect = new Effect(wctx.getScriptSession());
		effect.puff(id);
	}

	public void puffEffect(String id, String options) {
		WebContext wctx = WebContextFactory.get();
		Effect effect = new Effect(wctx.getScriptSession());
		effect.puff(id, "{" + options + "}");
	}

	public void pulsateEffect(String id) {
		WebContext wctx = WebContextFactory.get();
		Effect effect = new Effect(wctx.getScriptSession());
		effect.pulsate(id);
	}

	public void pulsateEffect(String id, String options) {
		WebContext wctx = WebContextFactory.get();
		Effect effect = new Effect(wctx.getScriptSession());
		effect.pulsate(id, "{" + options + "}");
	}
}